pkg/object/meshcontroller/api/api_ingress.go

```diff
@@ -73,34 +74,24 @@ func (a *API) createIngress(w http.ResponseWriter, r *http.Request) {
 	pbIngressSpec := &v1alpha1.Ingress{}
 	ingressSpec := &spec.Ingress{}
 
-	ingressName, err := a.readIngressName(w, r)
-	if err != nil {
-		api.HandleAPIError(w, r, http.StatusBadRequest, err)
-		return
-	}
-	err = a.readAPISpec(w, r, pbIngressSpec, ingressSpec)
+	err := a.readAPISpec(w, r, pbIngressSpec, ingressSpec)
 	if err != nil {
 		api.HandleAPIError(w, r, http.StatusBadRequest, err)
 		return
 	}
-	if ingressName != ingressSpec.Name {
-		api.HandleAPIError(w, r, http.StatusBadRequest,
-			fmt.Errorf("name conflict: %s %s", ingressName, ingressSpec.Name))
-		return
-	}
 
 	a.service.Lock()
 	defer a.service.Unlock()
 
-	oldSpec := a.service.GetIngressSpec(ingressName)
+	oldSpec := a.service.GetIngressSpec(ingressSpec.Name)
 	if oldSpec != nil {
-		api.HandleAPIError(w, r, http.StatusConflict, fmt.Errorf("%s existed", ingressName))
+		api.HandleAPIError(w, r, http.StatusConflict, fmt.Errorf("%s existed", ingressSpec.Name))
 		return
 	}
 
 	a.service.PutIngressSpec(ingressSpec)
 
-	w.Header().Set("Location", r.URL.Path)
+	w.Header().Set("Location", filepath.Join(r.URL.Path, ingressSpec.Name))
```

Comment:
the result of `filepath.Join` is OS-dependent.

```suggestion
	w.Header().Set("Location", path.Join(r.URL.Path, ingressSpec.Name))
```

Reply:
Updated with a dedicate util

Reply:
This is fine, but I still propose to use the standard `path` package, because its document says:

> Package path implements utility routines for manipulating slash-separated paths.
> 
> The path package should only be used for paths separated by forward slashes, such as the paths in URLs.

Reply:
Sure, good advice. Updated